server/main.go

```diff
@@ -52,7 +52,8 @@ func main() {
 	http.HandleFunc("/check/", CheckServer)
 	s := &http.Server{
 		Addr:        os.Args[1],
-		ReadTimeout: 10 * time.Second,
+		ReadTimeout: 1 * time.Second,
```

Comment:
I agree, this change is not necessary to lower the timeout when running against a host that is not responding. The only change that is actually needed is "deadline := time.Now().Add(2 * time.Second)" I'll amend my pull request to reflect this change.